packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/invoke-model.ts

```diff
@@ -221,15 +221,18 @@ export class BedrockInvokeModel extends sfn.TaskStateBase {
     }
 
     if (this.props.guardrail) {
+      const isArn = this.props.guardrail.guardrailIdentifier.startsWith('arn:');
```

Comment:
The policy document was incorrect for the case when guardrailIdentifier is an ARN, so I added a condition to address this.